### DISCUSSION TITLE
How to consume node modules in vscode.dev, I want to integrate a webview which uses React in my vscode.dev extension.

### DISCUSSION THREAD

**Discussion by amitjoshi438** [microsoft/vscode-discussions#813] [Extension Development QnA] [answered]
How to consume node modules in vscode.dev, I want to integrate a webview which uses React in my vscode.dev extension.

**Comment by alefragnani** [comment, marked as answer]
Hi @amitjoshi438 ,

You can't consume Node.js modules in vscode.dev. Instead, web extensions are restricted by the browser sandbox and therefore have [limitations](https://code.visualstudio.com/api/extension-guides/web-extensions#web-extension-main-file) compared to normal extensions.

I would suggest you to take a look at [GitLens](https://github.com/gitkraken/vscode-gitlens) source, as it works on vscode.dev and has webviews using React.

Hope this helps
